internal/controller/controllers/crd_utils.go

```diff
@@ -80,3 +84,27 @@ func AddLabel(labels map[string]string, labelKey, labelValue string) map[string]
 	labels[labelKey] = labelValue
 	return labels
 }
+
+type UpdateStatusFunc func(ctx context.Context, obj client.Object, opts ...client.UpdateOption) error
+
+func UpdateStatus(
+	log logrus.FieldLogger,
+	updateFunc UpdateStatusFunc,
+	ctx context.Context,
```

Comment:
nit: `ctx` by convention should be the first argument. From [the docs](https://golang.org/pkg/context/):

"Do not store Contexts inside a struct type; instead, pass a Context explicitly to each function that needs it. The Context should be the first parameter, typically named ctx:"